### PR TITLE
Update dependencies to avoiding pinning to old Rust package versions

### PR DESCRIPTION
**Pull request by pjkundert**
o Specify a "1.0" version for syn and quote, to allow upstream packages
  to specify newer versions; major/minor version compatibility should
  be sufficiently stable.
o Update shrinkwraprs to not pin an old syn version.

### CODE REVIEW

**Comment by pjkundert** [comment]
Also, make sure you merge https://github.com/freesig/observability/pull/1, to avoid pinning observability to an old version of holochain-serialization, too.

File: crates/holochain_json_api/Cargo.toml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 [package]
 name = "holochain_json_api"
-version = "0.0.46"
+version = "0.0.47"
```
Comment (freesig) [inline crates/holochain_json_api/Cargo.toml:3]: There's a nix script for doing this version bump and publishing. Better to use that then manually bumping I think.

**Review by freesig — CHANGES_REQUESTED** [review]
I like the general idea of this but I think we should use the [release procedure](https://docs.holochain.love/docs/release/) instead of manually bumping crate versions. Not so sure about syn and quote not breaking because it's possible something under them breaks and then we have no way to change the version from holochain

**Comment by pjkundert** [comment]
Closed in favor of https://github.com/holochain/holochain-serialization/pull/36

**Pull request closed without merging**